samples/log_callback/enclave/enc.c

```diff
@@ -30,24 +30,38 @@ void enclave_customized_log(
      * the host. The context might be used, for example, to transfer a
      * shared/public/private secret, that may be used to encrypt the log
      * message.
-     *
-     * NOTE: Do not use operations based on OE's host filesystem support
-     * in this function, they do not work consistently. If used, the
-     * logs generated after oe_terminate_enclave() cause the program to crash
-     * with a segmentation fault (issue #4349).
      */
-    OE_UNUSED(context);
+    fprintf((FILE*)context, "%s\n", modified_log);
+}
+
+FILE* enc_logfile = NULL;
+
+void enclave_set_log_callback(const char* abs_filepath)
```

Comment:
Follow OE https://github.com/openenclave/openenclave/blob/master/docs/Community/DevelopmentGuide.md#coding-conventions:

> DON'T use abbreviations unless they are already well-known terms known by users (e.g., "app", "info"), or are already required for use by developers (e.g., "min", "max", "args"). Examples of bad use would be num_widgets instead of widget_count, and opt_widgets instead of option_widgets or optional_widgets.
> DO use const and static and visibility modifiers to scope exposure of variables and methods as much as possible.
> The single underscore (_ ) is reserved for local definitions (static, file-scope definitions). e.g., static oe_result_t _parse_sgx_report_body(..).

```suggestion
static FILE* _enclave_logfile = NULL;

static void _enclave_set_log_callback(const char* absolute_filepath)
```